tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -1656,14 +1656,16 @@ void CGMSHLSLRuntime::AddHLSLFunctionInfo(Function *F, const FunctionDecl *FD) {
       funcProps->ShaderProps.GS.instanceCount = 1;
   }
 
+  // set isCs
+  for (auto *Attr : FD->specific_attrs<HLSLShaderAttr>()) {
+    if (Attr->getStage() == "compute") {
+      funcProps->shaderKind = DXIL::ShaderKind::Compute;
+      isCS = true;
+    }   
+  }
+
```

Comment:
This is correct now, but I think it's unnecessary. The isCS assignment present before was setting it inappropriately, but the `SetStageFlag` and `SetShaderKind` lambdas that get called by the `ParseShaderStage` lambda above will set both the shaderKind and isCS appropriately and I don't know of anything between there and here that would change that

Reply:
Prior to the addition of those lambdas, the assignment of isCS and other flags was strewn throughout this code. I'm not sure of the exact steps, but I think the assignment this code is trying to replace got left in somehow needlessly